### PR TITLE
TryFrom<GeoJson> for Geometry/Feature/FeatureCollection

### PR DESCRIPTION
**Pull request by rory**
Ergonomically 'unwrap' the GeoJson enum

This is a replacement for #140 without the extra dependency

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -44,4 +44,6 @@ pub enum Error {
     ExpectedArrayValue(String),
     #[error("Expected an owned Object, but got `{0}`")]
     ExpectedObjectValue(Value),
+    #[error("Cannot convert GeoJson requested enum variant")]
+    GeoJsonWrongType,
```
Comment (michaelkirk) [inline src/errors.rs:48]: Can we use the existing `ExpectedType` instead? 

Or if there's a reason not to, can we add similar `expected` vs. `actual` so we can get a good error message like `ExpectedType` has?

**Comment by rory** [comment]
The error type was one part I was unsure on, and I have no attachement to the current (new) error. Changing it to that would be fine by me. ☺

**Comment by rory** [comment]
I have amended the commit to use that Error type & force pushed. Is this OK to be merged?

**Review by michaelkirk — APPROVED** [review]
Thanks @rory - this LGTM, and seems uncontroversial, but I'm going to leave it open few a few days in case one of the longer-term maintainers wants to weigh in.

**Review by frewsxcv — APPROVED** [review]

**Comment by michaelkirk** [comment]
Thanks @rory!

**Pull request merged**